### PR TITLE
Add getWorkerAddresses and invalidate cache when registering worker

### PR DESCRIPTION
**Pull request by LuQQiu**
A loading cache is used for caching worker info list results. In this way, the overhead to construct worker info list is reduced but the worker info list result is not up to date. 

Add a getWorkerAddresses() method which returns the most up-to-date live worker addresses.
Add an invalidate cache in registering worker to make sure getWorkerInfoList() call catchs up with the worker changes.

### CODE REVIEW

**Comment by LuQQiu** [comment]
alluxio-bot, check this please

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 1]
```diff
@@ -463,6 +463,19 @@ public long getUsedBytes() {
     return workerInfoList;
   }
 
+  @Override
+  public Set<WorkerNetAddress> getWorkerAddresses() throws UnavailableException {
+    if (mSafeModeManager.isInSafeMode()) {
+      throw new UnavailableException(ExceptionMessage.MASTER_IN_SAFEMODE.getMessage());
+    }
+    Set<WorkerNetAddress> workerAddresses = new HashSet<>();
```
Comment (bf8086) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: (nit) Initialize with initial capacity to be the number of workers.

**Comment by LuQQiu** [comment]
@bf8086 Tested, PTAL, thanks

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
alluxio-bot, cherry-pick this to branch-2.2 please

**Pull request merged**